doc/user/content/sql/join.md

```diff
@@ -53,6 +60,45 @@ Instead, `JOIN`s work over the available history of both streams, which
 ultimately provides an experience more similar to an RDBMS than other streaming
 platforms.
 
+### `LATERAL` subqueries
+
+When a subquery appears on the right-hand side of a join, it can optionally be
+preceded by the `LATERAL` keyword. This permits the subquery to refer to columns
+defined by the left-hand side of the join. Normally, a subquery only has access
+to the columns within its own context.
+
+Table functions may also be preceded by the `LATERAL` keyword, although in this
+case the keyword is optional. Table function invocations always have implicit
+access to the columns defined by the left-hand side of the join.
+
+When `LATERAL` behavior is active, the right-hand relation is recomputed for
```

Comment:
nit: Could we say something like...

> **LATERAL** expressions (i.e. the right-hand relation) are recomputed for each row in the left-hand relation.

Talking about "behavior being active" makes it sound like a setting, which feels odd to me when talking about SQL statements. Changing this isn't worth more than 10 seconds of thought, though.